File: src/main/java/kz/superkassa/tests/framework/db/DatabaseClient.java
package ru.superkassa.tests.framework.db;

import com.zaxxer.hikari.HikariConfig;
import com.zaxxer.hikari.HikariDataSource;
import ru.superkassa.tests.framework.config.TestConfig;

import javax.sql.DataSource;
import java.sql.Connection;
import java.sql.SQLException;

public final class DatabaseClient implements AutoCloseable {
    private final HikariDataSource dataSource;

    public DatabaseClient(TestConfig config) {
        HikariConfig hikariConfig = new HikariConfig();
        hikariConfig.setJdbcUrl(config.databaseJdbcUrl());
        hikariConfig.setUsername(config.databaseUsername());
        hikariConfig.setPassword(config.databasePassword());
        hikariConfig.setReadOnly(true);
        hikariConfig.setMaximumPoolSize(3);
        this.dataSource = new HikariDataSource(hikariConfig);
    }

    public DataSource dataSource() {
        return dataSource;
    }

    public Connection connection() throws SQLException {
        return dataSource.getConnection();
    }

    @Override
    public void close() {
        dataSource.close();
    }
}

File: src/main/kotlin/kz/superkassa/tests/framework/db/DatabaseClient.kt
package kz.superkassa.tests.framework.db

import com.zaxxer.hikari.HikariConfig
import com.zaxxer.hikari.HikariDataSource
import kz.superkassa.tests.framework.config.TestConfig
import java.sql.Connection
import javax.sql.DataSource

@Suppress("unused")
class DatabaseClient(config: TestConfig) : AutoCloseable {
    private val dataSource: HikariDataSource = HikariDataSource(
        HikariConfig().apply {
            jdbcUrl = config.databaseJdbcUrl
            username = config.databaseUsername
            password = config.databasePassword
            isReadOnly = true
            maximumPoolSize = 3
        },
    )

    fun dataSource(): DataSource = dataSource

    fun connection(): Connection = dataSource.connection

    override fun close() {
        dataSource.close()
    }
}
